### PR TITLE
Accept function for option dependsOn

### PR DESCRIPTION
**Pull request by andytuba**
Enable complex dependsOn behavior like values of an enum

attn: @KetillG

### CODE REVIEW

File: lib/core/module.js [hunk 1]
```diff
@@ -53,7 +53,7 @@ export type BooleanOption = {|
 	keywords?: Array<string>,
 	value: boolean,
 	bodyClass?: boolean | string,
-	dependsOn?: string,
+	dependsOn?: string | function,
```
Comment (erikdesjardins) [inline lib/core/module.js]: `dependsOn?: string | () => boolean,`

**Comment by erikdesjardins** [comment]
I want to do a bit more with this: require dependsOn to be a function (so we catch misspellings etc.) and possibly subsume noConfig, but I'll just push to this branch to take care of that (unless you want to)

File: lib/core/module.js (hunk 1 above)
Comment (andytuba) [inline lib/core/module.js]: oh duh i have to specify the function signature

**Comment by andytuba** [comment]
It would be nice to preserve the simple legacy behavior of `dependsOn` as "option whose value is truthy". The settingsConsole framework would throw an error either way, if it's a plain string indirection or directly referenced within a function.  ... Would flow catch a misspelled direct reference to an option?

Adding dependsOn to bodyClass sounds great.

Go ahead with pushing to this branch.

**Comment by andytuba** [comment]
:warning: TODO: `settingsConsole` should call `onOptionChanged` for more option types, e.g. enum's, and expand `onOptionChanged` to also handle the function type.

**Pull request closed without merging**